### PR TITLE
add contextInfoContainer calculation

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: src/alt-util.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace altintegration {
+
+uint256 CalculateContextInfoContainerHash(const PopData& popData,
+                                          const BlockIndex<AltBlock>& prevBlock,
+                                          const AltChainParams& params) {
+  auto firstPreviousKeystone =
+      prevBlock.getAncestor(getFirstPreviousKeystoneHeight(
+          prevBlock.getHeight() + 1, params.getKeystoneInterval()));
+  auto secondPreviousKeystone =
```
Comment (Warchant) [inline src/alt-util.cpp:18]: this can be nullptr if ancestor is not found (for example VBK10->getAncestor(100) or VBK10->getAncestor(-100))

Comment (Warchant) [reply src/alt-util.cpp:18]: If keystone is nullptr, then make it equal to `000..000`
```
auto zeroHash = prevBlock.getHash().fill(0);
```

**Review by Warchant — CHANGES_REQUESTED** [review]

File: src/alt-util.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  auto atvMerkleRoot =
+      PayloadsMerkleTree<ATV>(map_get_id(popData.atvs)).getMerkleRoot();
+  auto vtbMerkleRoot =
+      PayloadsMerkleTree<VTB>(map_get_id(popData.vtbs)).getMerkleRoot();
+  auto vbkMerkleRoot =
+      PayloadsMerkleTree<VbkBlock>(map_get_id(popData.context)).getMerkleRoot();
+
+  uint256 left = sha256twice(atvMerkleRoot, vtbMerkleRoot);
+  uint256 right = sha256twice(vbkMerkleRoot, vbkMerkleRoot);
```
Comment (Warchant) [inline src/alt-util.cpp:41]: pls change the order
```
(vbk, vtb)  (atv, context)
```

Comment (Mr-Leshiy) [reply src/alt-util.cpp]: fixed

File: src/alt-util.cpp (hunk 1 above)
Comment (Mr-Leshiy) [reply src/alt-util.cpp]: fixed

**Pull request merged**